### PR TITLE
chore: clean up last couple of utility class usages

### PR DESCRIPTION
**Pull request by schakraborty22**
### Summary:
This PR is to clean up utilities classes
1) u-margin-top-xl-mobile
2) u-theme-typography-body-text-xs-bold
Updated with tailwind classes and css modules styles.

### Test Plan:
1) run storybook.
2) run test.
3) run chromatic to verify UI changes.

### CODE REVIEW

File: src/components/Section/Section.stories.tsx [hunk 1]
```diff
@@ -34,11 +34,7 @@ export const Center: StoryObj<Args> = {
     ),
     title: 'Section Title',
     description: (
-      <Text
-        as="p"
-        className={styles['u-theme-typography-body-text-xs-bold']}
-        size="sm"
-      >
+      <Text as="p" className={styles['section__text']} size="sm">
```
Comment (jinlee93) [inline src/components/Section/Section.stories.tsx:37]: Does `section__text` exist in the stylesheet?

File: .storybook/pages/ProjectOverview/ProjectOverview.tsx [hunk 2]
```diff
@@ -31,9 +33,14 @@ export interface Props {
...
   activeIndex?: number;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
 }
 
-export const ProjectOverview = ({ activeIndex = 0 }: Props) => {
+export const ProjectOverview = ({ activeIndex = 0, className }: Props) => {
+  const textClassName = clsx(styles['projectOverview__text'], className);
```
Comment (jinlee93) [inline .storybook/pages/ProjectOverview/ProjectOverview.tsx:43]: nit: bem syntax should be `project-overview__text`

File: .storybook/pages/ProjectOverview/ProjectOverview.module.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@import '../../../../edu-design-system/src/design-tokens/mixins.css';
+
+/**
+ *  Utility classes to apply typography usage mixins to Project Overview in body text xs bold
+ */
+.projectOverview__text {
```
Comment (jinlee93) [inline .storybook/pages/ProjectOverview/ProjectOverview.module.css:6]: nit: bem syntax should be `project-overview__text`

File: .storybook/pages/ProjectOverview/ProjectOverview.tsx [hunk 4]
```diff
@@ -234,7 +241,7 @@ export const ProjectOverview = ({ activeIndex = 0 }: Props) => {
             <Text as="div" className="!mb-4">
               <p>
                 Students individually develop a follow up question that builds
-                from their group’s experiment and describe their follow up
+                from their group’s experiments and describe their follow up
```
Comment (dierat) [inline .storybook/pages/ProjectOverview/ProjectOverview.tsx:244]: Was this change intentional?

File: .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.module.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+/*------------------------------------*\
+  # TYPOGRAPHY USAGE UTILITY CLASS
+\*------------------------------------*/
+
+/**
+ *  Utility classes to apply typography usage mixins to elements for Course Planner Step module
+ */
+
+.coursePlannerStep1__text{
```
Comment (jinlee93) [inline .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.module.css:11]: nit: bem syntax is `course-planner-step1__text`

File: .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx [hunk 6]
```diff
@@ -18,84 +20,96 @@ import {
...
+export interface Props {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+}
+
+export const CoursePlannerStep1 = (className: Props) => {
+  const textClassName = clsx(
+    `${styles['coursePlannerStep1__text']} !mb-2`,
```
Comment (jinlee93) [inline .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx:34]: nit: bem syntax is `course-planner-step1__text`

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx [hunk 7]
```diff
@@ -30,6 +32,13 @@ import TableCard from '../../recipes/TableCard';
 
 import EmptyImage from '../../static/hand-pencil.svg';
 
+export interface Props {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+}
+
```
Comment (dierat) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:41]: You don't need to add the `className` props because these are "pages"; they won't be reused with people passing in styles. That pattern of having a `className` prop only applies to components that are used in various contexts and may need to be customized.

Comment (dierat) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:41]: You don't need to add the `className` props because these are "pages"; they won't be reused with people passing in styles. That pattern of having a `className` prop only applies to components that are used in various contexts and may need to be customized.

```suggestion
```

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx [hunk 8]
```diff
@@ -262,7 +271,7 @@ const StandardsRows = [
   },
 ];
 
-export const CoursePlannerEdit = () => {
+export const CoursePlannerEdit = (className: Props) => {
```
Comment (dierat) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:274]:
```suggestion
export const CoursePlannerEdit = () => {
```

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx [hunk 9]
```diff
@@ -499,7 +508,7 @@ export const CoursePlannerEdit = () => {
               // above it only when the flex-direction is column, so
               // this utility class was written with a media query
               // that matches the behavior of .page-header
-              <ButtonGroup className="u-margin-top-xl-mobile">
+              <ButtonGroup className={buttonClassName}>
```
Comment (dierat) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:511]: You can just use the your imported class name here directly and skip the variable.
```suggestion
              <ButtonGroup className={styles['button--mobile']}>
```

File: .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx [hunk 10]
```diff
@@ -18,84 +20,96 @@ import {
...
-  <>
-    <Panel className="!mb-8" variant="squared">
+export interface Props {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+}
+
+export const CoursePlannerStep1 = (className: Props) => {
```
Comment (dierat) [inline .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx:32]: This is another "page"; we don't need the `className` prop.
```suggestion
export const CoursePlannerStep1 = () => {
```

File: .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx [hunk 11]
```diff
@@ -18,84 +20,96 @@ import {
...
+  const textClassName = clsx(
+    `${styles['coursePlannerStep1__text']} !mb-2`,
+    className,
+  );
```
Comment (dierat) [inline .storybook/pages/CoursePlannerStep1/CoursePlannerStep1.tsx:36]:
```suggestion
  const textClassName = clsx(
    styles['coursePlannerStep1__text'],
    "!mb-2",
  );
```

File: .storybook/pages/ProjectOverview/ProjectOverview.tsx (hunk 4 above)
Comment (schakraborty22) [reply .storybook/pages/ProjectOverview/ProjectOverview.tsx]: Sorry mistyped mug fell on keyboard !

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx (hunk 7 above)
Comment (schakraborty22) [reply .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:41]: Thanks @dierat !

**Review by dierat — APPROVED** [review]
Looks great, thanks!
